core/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/OAuth2AccessTokenAuthenticationToken.java

```diff
@@ -60,9 +61,23 @@ public OAuth2AccessTokenAuthenticationToken(RegisteredClient registeredClient,
 		this.accessToken = accessToken;
 	}
 
+	/**
+	 * Constructs an {@code OAuth2AccessTokenAuthenticationToken} for public clients, using the provided parameters.
+	 *
+	 * @param clientId the client id
+	 * @param accessToken the access token
+	 */
+	public OAuth2AccessTokenAuthenticationToken(String clientId, OAuth2AccessToken accessToken) {
```

Comment:
Please remove this constructor as the existing constructor may be used. For the `Authentication clientPrincipal` argument, use `new OAuth2ClientAuthenticationToken(RegisteredClient)`